File: Day46/PrimeNoOfSetBit_BinRep.py
class Solution:
    def countPrimeSetBits(self, left: int, right: int) -> int:
        prime_set = {2, 3, 5, 7, 11, 13, 17, 19}
        count = 0
        
        for num in range(left, right + 1):
            set_bits = bin(num).count('1')
            
            if set_bits in prime_set:
                count += 1
                
        return count

def main():
    print("--- 762. Prime Number of Set Bits (Python) ---")
    
    try:
        left = int(input("Enter the left bound (integer): "))
        right = int(input("Enter the right bound (integer): "))
    except ValueError:
        print("Invalid integer input for bounds.")
        return

    if left > right:
        print("Error: Left bound must be less than or equal to the Right bound.")
        return

    solution = Solution()
    result = solution.countPrimeSetBits(left, right)

    print(f"\nInput: left = {left}, right = {right}")
    print(f"Output (Count of numbers with prime set bits): {result}")

if __name__ == "__main__":
    main()

File: Day46/PrimeNoOfSetBit_BinRep.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class PrimeNoOfSetBit_BinRep {
    public int countPrimeSetBits(int left, int right) {
        Set<Integer> primes = new HashSet<>(Arrays.asList(
            2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31
        ));

        int count = 0;
        for (int i = left; i <= right; i++) {
            int bits = Integer.bitCount(i);
            if (primes.contains(bits)) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.println("--- 762. Prime Number of Set Bits (Java) ---");
        
        System.out.print("Enter the left bound (integer): ");
        int left = scanner.nextInt();
        
        System.out.print("Enter the right bound (integer): ");
        int right = scanner.nextInt();

        PrimeNoOfSetBit_BinRep solution = new PrimeNoOfSetBit_BinRep();
        int result = solution.countPrimeSetBits(left, right);

        System.out.println("\nInput: left = " + left + ", right = " + right);
        System.out.println("Output (Count of numbers with prime set bits): " + result);

        scanner.close();
    }
}